ci/generate_version_tag.py

```diff
@@ -0,0 +1,76 @@
+"""Version Tag Generator
+
+This script is designed to parse a list of commits written according to the
+conventional commit guidelines https://www.conventionalcommits.org/en/v1.0.0/.
+Based on the type of each commit either a patch or a minor version is updated
+from the original tag provided with respect to semantic versioning
+https://semver.org/.
+
+If a script encounters a breaking change in the changelog it will throw an error
+in order to make sure major releases are expected and explicit. There is a parameter
+passed to the script to explicitly trigger a major version increase.
+
+Prerelease tags (e.g. -alpha) are passed manually, if required by the build state.
+They will be appended to the next incremented version. A run when the previous version
+provided was a pre-release would finalize it e.g. (1.0.1-alpha -> 1.0.1)
+"""
+
+import argparse
+from typing import Tuple
+
+import semver
+
+
+def main(changes: str, old_tag: str, prerelease_tag: str, major_release: bool):
+    """
+    Generate new version tag.
+
+    Args:
+        changes: list of newline-separated change titles
+        old_tag: previous version tag e.g. 1.0.1
+        prerelease_tag: add a pre-relrease tag to the next version e.g -alpha
+        major_release: boolean to override and trigger a major release
+    """
+    ver = semver.VersionInfo.parse(old_tag)
+    if major_release:
+        new_ver = ver.bump_major()
+    else:
+        if not changes:
+            raise Exception("No changes since the last release!")
+        # Breaking changes are of form <tag>!: <description>
+        # To avoid automatically publishing major version they need to be
+        # explicitly specified
+        breaking = [c for c in changes.split("\n") if "!:" in c]
```

Comment:
We need to make sure that no one has `!:` in their PR name accidentally (i guess that's pretty unlikely)
Other solution could be to do `c.split(':', 1)[0].endswith('!')`

Reply:
Yeah, doesn't look like it's spotted by the PR title checker so better be safe.